autorest/azure/environments.go

```diff
@@ -126,5 +135,19 @@ func EnvironmentFromName(name string) (Environment, error) {
 	if !ok {
 		return env, fmt.Errorf("autorest/azure: There is no cloud environment matching the name %q", name)
 	}
+
+	if strings.EqualFold(name, "AZURESTACKCLOUD") == true {
```

Comment:
There seems like enough logic here that it calls for its own constructor. I'm thinking that maybe we just should have "AzureStackCloud" be one of the named environments.